code/string-calculator/viewmodel/src/main/java/ru/unn/agile/calc/viewmodel/StringCalcViewModel.java

```diff
@@ -77,4 +129,27 @@ private boolean isCalculationDisabled() {
     public BooleanProperty calculationDisabledProperty() {
         return calculationDisabled;
     }
+
+    public StringProperty logsProperty() {
+        return logs;
+    }
+
+    public final String getLogs() {
+        return logs.get();
+    }
+
+    public final List<String> getLog() {
+        return logger.getLog();
```

Comment:
What should happend if logger is null? (Default constructor of ViewModel)

Reply:
added NPE check